src/libutil/util.cc

```diff
@@ -1831,4 +1831,26 @@ void commonChildInit(Pipe & logPipe)
     close(fdDevNull);
 }
 
+/* Given two pointers to c strings, return a pointer to the start of the first
+ * non-equivalent part of the first string. This is a non-owning reference to
+ * data owned by a, and must not outlive a. */
+const std::string_view removeStartingOverlap(std::string & a_ref, std::string & b_ref)
+{
+    auto a = std::string_view(a_ref);
+    auto b = std::string_view(b_ref);
+
+    auto start = 0;
+    auto end = a.length() > b.length() ? a.length() : b.length();
+
+    for (auto ap = a.begin(), bp = b.begin(),
+              ae = a.end(),   be = b.end();
+         ap < ae && bp < be;
+         ++ap, ++bp, ++start
+    ) {
+        if (*ap != *bp) break;
+    }
+
+    return a.substr(start, end - start);
```

Comment:
Don’t you want 
```suggestion
    return a.substr(start);
```
?

I think `a.substr(start, end-start)` will be too short if `a.length()>b.length()` (which might not happen here, but still)

Reply:
I’m not sure, but maybe you can also rewrite this as

```suggestion
    return std::string_view(ap, a.end());
```

in which case you won’t need `start` anymore